docs/website/docs/guides/ml-frameworks/pytorch.md

```diff
@@ -5,197 +5,243 @@ tags:
-as well as models defined using [`functorch`](https://pytorch.org/functorch/).
+!!! note "Note - under development"
+    The PyTorch ecosystem is evolving rapidly. APIs and workflows are expected
+    to change.
 
-``` mermaid
+## :octicons-book-16: Overview
+
+[SHARK-Turbine](https://github.com/nod-ai/SHARK-Turbine) offers a tight
+integration between compatible versions of IREE,
+[torch-mlir](https://github.com/llvm/torch-mlir), and
+[PyTorch](https://pytorch.org/).
+
+- [x] Seamless integration with standard PyTorch workflows
+- [x] Deployment support for running PyTorch models on cloud and edge devices
+- [x] General purpose model compilation and execution tools
+
+Both just-in-time (JIT) and ahead-of-time (AOT) workflows are supported:
+
+```mermaid
 graph LR
-  accTitle: PyTorch to runtime deployment workflow overview
+  accTitle: PyTorch integration overview
   accDescr {
-    Programs start as either PyTorch nn.Module or functorch programs.
-    Programs are imported into MLIR as either StableHLO, TOSA, or Linalg.
-    The IREE compiler uses the imported MLIR.
-    Compiled programs are used by the runtime.
+    PyTorch programs can be optimized within a Python session with
+    SHARK-Turbine's just-in-time tools.
+    PyTorch programs can be exported out of Python to native binaries using
+    SHARK-Turbine's ahead-of-time export toolkit.
   }
 
-  subgraph A[PyTorch]
-    direction TB
-    A1[nn.Module]
-    A2[functorch]
+  subgraph Python
+    pytorch(PyTorch)
+    subgraph turbine [SHARK-Turbine]
+      jit("Eager execution (JIT)")
+      aot("Export toolkit (AOT)")
+    end
 
-    A1 --- A2
+    pytorch --> jit
+    jit --> pytorch
+    pytorch --> aot
   end
 
-  subgraph B[MLIR]
-    direction TB
-    B1[StableHLO]
-    B2[TOSA]
-    B3[Linalg]
-
-    B1 --- B2
-    B2 --- B3
+  subgraph Native
+    binary(["binary (.vmfb)"])
   end
 
-  C[IREE compiler]
-  D[Runtime deployment]
-
-  A -- torch_mlir --> B
-  B --> C
-  C --> D
+  aot -.-> binary
 ```
 
 ## :octicons-download-16: Prerequisites
 
-1. Install IREE packages, either by
-    [building from source](../../building-from-source/getting-started.md#python-bindings)
-    or from pip:
+Install Turbine and its requirements:
+
+``` shell
+python -m pip install shark-turbine
+```
 
-    === "Stable releases"
+## :octicons-flame-16: Just-in-time (JIT) execution
 
-        Stable release packages are
-        [published to PyPI](https://pypi.org/user/google-iree-pypi-deploy/).
+Just-in-time integration allows for Python code using TorchDynamo to optimize
+PyTorch models/functions using IREE, all within an interactive Python session.
 
-        ``` shell
-        python -m pip install \
-          iree-compiler \
-          iree-runtime
-        ```
+<!-- TODO(scotttodd): mention targets like AMD GPUs when supported
+                      https://github.com/nod-ai/SHARK-Turbine/issues/94 -->
+
+``` mermaid
+graph TD
+  accTitle: PyTorch JIT workflow overview
```

Comment:
Starting to hit the limits of what https://mermaid.js.org/ can produce (expressiveness and visual polish). Might convert some of these to https://github.com/excalidraw/excalidraw (which the main architecture diagram in the README already uses)